### DISCUSSION TITLE
"Install" should NOT "upgrade everything" without warning - now what?

### DISCUSSION THREAD

**Discussion by drawkkwast** [Homebrew/discussions#3419] [Everyday usage] [answered]
I'm on a MacBook Pro M1 and ran `$ brew install python`

To my horror, this resulted in EVERYTHING getting upgraded, including PostgreSQL from `13.7` to `14.4`

Who would expect that installing python would automatically and without warning, do a major version upgrade to PostgreSQL, which breaks it???  And no warning or confirmation messages???  If I would have known I had to "pin" it, or if I would have been told "Warning: you are about to upgrade PostgreSQL because you want to install python" I would NOT have run the python install  ...this is insane IMO, because who would expect this behavior?

So then I run `$ brew postgresql-upgrade-database` which fails.

Exactly how do a do a manual/full uninstall of the 2 versions of PostgreSQL under homebrew, and then do a clean install?

**Comment by drawkkwast** [comment, marked as answer]
Actually, I'm done with homebrew...  I've uninstalled and am now running the Postgres.app

This is now the only way Heroku recommends to instal psql (because apparently homebrew has pissed off thousands of devs with this "feature" that silently and unexpected breaks your databases)

**Comment by SMillerDev** [comment]
> Who would expect that installing python would automatically and without warning, do a major version upgrade to PostgreSQL, which breaks it???

Better framing: "who would expect python updates to also ensure its dependencies are able to work with that python version?" To which the answer is: everyone would expect their system to keep working after an update.

https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff

**Reply by drawkkwast** [reply]
> everyone would expect their system to keep working after an update

Better framing: "everyone would expect their **_PostgreSQL_** to keep working after an update" - which it did NOT!

You want to start an argument!?!?  Let's argue...  I have seen literally thousands of devs all over the internet complaining about this, and you in your vast arrogance won't even consider adding a few lines of code to simply output to the user what packages will be upgraded, giving them an option to cancel?!?

Go ahead and reply.  Let's do this, right here, right now.  Tell me exactly how I'm wrong, and why you refuse to do something like the above suggestion.  How would adding a **simple warning** be the end of the world for you?  I know a few thousand pissed off homebrew users want to know.

**Reply by SMillerDev** [reply]
> and why you refuse to do something like the above suggestion.

Because nobody is paying me for it and I know to run `brew outdated` if I want to know what will be updated.

**Reply by drawkkwast** [reply]
> Because nobody is paying me for it

...and folks, we have our answer:  They don't care.  Got it!  Thank you for at least being honest.

**Reply by SMillerDev** [reply]
Well, clearly you care just enough to be rude to volunteers but not enough to actually fix it. It's the beauty of open source, everyone can make fixes for things they see as a problem.

**Reply by drawkkwast** [reply]
I'll give you the **_exact_** same answer you gave me: "Because nobody is paying me for it!"

**Reply by SMillerDev** [reply]
With the key difference being: Why would anyone pay you to fix your own problems?

**Reply by drawkkwast** [reply]
It's NOT my software.  This is a project you work on, NOT me.  Exactly what are you hoping to accomplish here?  Your point is what???  You said above that you're NOT paid and you don't care.  Okay.  That's where this ends.  But you'd rather what, attack one of your former users publicly, tell him he should fix homebrew himself because you don't care, and then publicly make a fool of yourself with very obviously hypocritical statements...  while embarrassing yourself and the whole group you work with???

Are you sure you want to continue this conversation?  ...because you are being so ridiculous here that this thread is going to get noticed.

**Reply by SMillerDev** [reply]
> It's NOT my software. This is a project you work on, NOT me.

I see you don't understand open source software. I guess all I can do is leave you with a quote from the Homebrew license:

> THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.

**Reply by drawkkwast** [reply]
EXACTLY - Don't expect it to work!  Go it!  Anything else other than:

1. You're NOT paid.
2. You don't care.
3. Don't expect it to work.

???

**Reply by p-linnane** [reply]
The language used in the OP was needlessly aggressive, which doomed this discussion from the start. Asking for help and using a positive or neutral tone is going to get you a lot further than calling things you don't agree with "insane" or flaming. 

The entire point of freely provided open source software is so that anyone can contribute and discuss with the community. When I find something that isn't working correctly, I open a PR with a fix. And if I can't fix it myself, I'll open an issue or discussion so that the community can collaborate with me. 

This is all done for free by volunteers who contribute because they love the project. If you are in a position where your PostgreSQL install is so critical you're willing to act like this publicly, you should consider purchasing their commercial professional support services instead of using community-maintained solutions: https://www.postgresql.org/support/professional_support/northamerica/
